### DISCUSSION TITLE
Is it possible to get `TextFrame`'s shape information?

### DISCUSSION THREAD

**Discussion by BiDuang** [ShapeCrawler/ShapeCrawler#595] [Q&A]
How to get a `ITextFrame` object's shape information. For example, `DocumentFormat.OpenXml.Presentation.TextBody`?

**Comment by BiDuang** [comment]
I also noticed that there is a `private readonly ITextFrameContainer textFrameContainer;`, it looks like containing the parent shape of the textframe, but it's private and cannot access from outside. How about setting it to `public`?

**Reply by BiDuang** [reply]
Also, `GetAllTextFrames` may should return `SCTextFrame` instead of `ITextFrame`?

**Reply by AdamShakhabov** [reply]
Currently, there is no capability to access the underlying `DocumentFormat.OpenXml.Presentation.TextBody` of a text frame. If needed, we could add a new property, like `ITextFrame.SDKTextBody`. However, this property should return a _copy_ of `TextBody` to avoid breaking encapsulation.

Be aware that significant refactoring changes have been merged into the `master` branch.

**Reply by BiDuang** [reply]
I think `IShape` should add a property called `OpenXMLElement` that allows others can access the origin element.

**Reply by AdamShakhabov** [reply]
@BiDuang, a new `IShape.SDKOpenXmlElement` was added. Please check this in the latest version [**v0.50.1**](https://github.com/ShapeCrawler/ShapeCrawler/releases)
